File: python3/jac_format/join_path.py
def join_path(row_path, column_path):
    if column_path is None:
        raise Exception("Column path is empty!")
    if row_path is None:
        raise Exception("Row path is empty")
    
    column_path = column_path.strip()
    row_path = row_path.strip()
    
    if len(row_path) > 0 and row_path[-1] == ".":
        row_path = row_path[0:-1]
    if len(column_path) > 0 and column_path[-1] == ".":
        column_path = column_path[0:-1]
    if len(row_path) > 0 and row_path[0] == ".":
        row_path = row_path[1:]
    if len(column_path) and column_path[0] != ".":
        column_path = "." + column_path
        
    if row_path == "":
        return column_path[1:] if len(column_path) > 1 else "."
    if column_path == ".":
        return row_path

    return row_path+column_path

File: javascript/lib/utils/join-path.js
function joinPath(rowPath, columnPath) {
  if (!rowPath) throw new Error(`Column path is empty!`)
  if (!columnPath) throw new Error(`Row path is empty!`)

  rowPath = rowPath.trim()
  columnPath = columnPath.trim()

  // Remove dots from beginning and end of rowPath, make sure columnPath has
  // a dot in front of it
  if (rowPath.endsWith(".")) rowPath = rowPath.slice(0, -1)
  if (rowPath.endsWith(".")) columnPath = columnPath.slice(0, -1)
  if (rowPath.startsWith(".")) rowPath = rowPath.slice(1)
  if (!columnPath.startsWith(".")) columnPath = "." + columnPath

  if (rowPath === "") return columnPath.slice(1) || "."
  if (columnPath === ".") return rowPath

  return rowPath + columnPath
}

module.exports = joinPath
